### ISSUE TITLE
Artifact downloaded but 'No dependency graph files found to submit' when using `download-and-submit`

### ISSUE THREAD

**Issue by Pil0tXia** [gradle/actions#196]
I tried [Integrating `dependency-submission` and `dependency-review-action` for pull requests from public forked repositories](https://github.com/gradle/actions/blob/d211a39090cba0cdce485f2a0b1f28f39ccda0c9/docs/dependency-submission.md#integrating-dependency-review-action-for-pull-requests-from-public-forked-repositories), the workflow files are as follows:

- https://github.com/Pil0tXia/eventmesh/blob/2f7c34ab75a92871d80b32b841a591ec6ffd846a/.github/workflows/code-scanning.yml
- https://github.com/Pil0tXia/eventmesh/blob/2f7c34ab75a92871d80b32b841a591ec6ffd846a/.github/workflows/dependency-graph.yml
- https://github.com/Pil0tXia/eventmesh/blob/2f7c34ab75a92871d80b32b841a591ec6ffd846a/.github/workflows/license.yml

However, neither `dependency-submission@v3` ([Commit](https://github.com/Pil0tXia/eventmesh/commit/2f7c34ab75a92871d80b32b841a591ec6ffd846a), [Actions log](https://github.com/Pil0tXia/eventmesh/actions/runs/8799849088/job/24149741831)) or `setup-gradle@v3` ([Commit](https://github.com/Pil0tXia/eventmesh/commit/df3532d5a707b9eb9d9ad8bf61dde33801ac329a), [Actions log](https://github.com/Pil0tXia/eventmesh/actions/runs/8800418554/job/24151556730)), the following log was output, and the gradle dependencies were not added to the dependency graph of the Repo (that is, the artifacts marked with `Detected by GitHub Dependency Graph Gradle Plugin`):

```
Artifact download completed successfully.
Downloading dependency-graph artifact dependency-graph_code_scanning-build-java.json to /home/runner/work/eventmesh/eventmesh
Found dependency graph files:
No dependency graph files found to submit.
```

When I rolled back to `gradle/gradle-build-action@v2`, it worked ([Commit](https://github.com/Pil0tXia/eventmesh/commit/b2ff90ed424050b5a93232cba31596487a6d36ff), [Actions log](https://github.com/Pil0tXia/eventmesh/actions/runs/8804550745/job/24165165445)), and the dependency graph of the Repo had been updated.

**Comment by bigdaz** [comment]
Thanks for your report. The dependency-graph is being generated and saved ([you can see the attached artifact here](https://github.com/Pil0tXia/eventmesh/actions/runs/8804409522)), but it appears that the `download-and-submit` part of the workflow is not working.

This functionality is difficult to verify with automated testing, so it's possible I introduced a bug as part of [this fix in v3.3.1](https://github.com/gradle/actions/pull/189).
Can you please try using `v3.3.0` (`gradle/actions/dependency-submission@v3.3.0`)?

**Comment by Pil0tXia** [comment]
@bigdaz

Hi! I've made changes in https://github.com/Pil0tXia/eventmesh/commit/b96139c0dfe605b7b6fbf7ac822cc705429b7095, it seems `v3.3.0` works well on [master branch](https://github.com/Pil0tXia/eventmesh/actions/runs/8814153002/job/24193390904) and [PR branch](https://github.com/Pil0tXia/eventmesh/actions/runs/8814209957/job/24193563339).

**Comment by bigdaz** [comment]
Thanks for letting me know. I'll try to get this fixed ASAP.
Once this issue is resolved (and a new release published) you should be able to switch back to using `v3`.

**Comment by bigdaz** [comment]
@Pil0tXia Thanks again for the thorough report: it helped me get this fixed quickly.
I've released `v3.3.2` with the fix so you should be able to switch back to using `v3`.
